templates/http/default.conf.j2

```diff
@@ -51,7 +51,12 @@ auth_request {{ item.value.auth_request_http }};
 
 server {
 {% if item.value.ssl is defined and item.value.ssl %}
-    listen {{ item.value.port }} ssl;
+{% if ( item.value.listen.ipv4 is defined and item.value.listen.ipv4 ) or ( item.value.listen.ipv4 is not defined ) %}
+    listen {{ item.value.listen.port }} ssl{% if item.value.listen.opts is defined and item.value.listen.opts|length %} {{ item.value.listen.opts|join(" ") }}{% endif %};
```

Comment:
Maybe we can remove some duplicate code by changing the `ssl` option into a conditional check, moving the `item.value.ssl` conditional check to only wrap around the `ssl` directives and then removing the listen block at https://github.com/nginxinc/ansible-role-nginx/pull/147/files#diff-ec7d84372203d5d7af253681f4c6fc67R78?

Reply:
Sounds good to me.
I'll change my PR.